dist/jstat.js

```diff
@@ -2799,6 +2810,363 @@ jStat.extend(jStat.laplace, {
+    pr_w = 0.0;
+
+  // if w is large then the second component of the
+  // integral is small, so fewer intervals are needed.
+
+  var wincr;
+  if (w > wlar)
+    wincr = wincr1;
+  else
+    wincr = wincr2;
+
+  // find the integral of second term of hartley's form
+  // for the integral of the range for equal-length
+  // intervals using legendre quadrature.  limits of
+  // integration are from (w/2, 8).  two or three
+  // equal-length intervals are used.
+
+  // blb and bub are lower and upper limits of integration.
+
+  var blb = qsqz;
+  var binc = (bb - qsqz) / wincr;
+  var bub = blb + binc;
+  var einsum = 0.0;
+
+  // integrate over each interval
+
+  var cc1 = cc - 1.0;
+  for (var wi = 1; wi <= wincr; wi++) {
+    var elsum = 0.0;
+    var a = 0.5 * (bub + blb);
+
+    // legendre quadrature with order = nleg
+
+    var b = 0.5 * (bub - blb);
+
+    for (var jj = 1; jj <= nleg; jj++) {
+      var j, xx;
+      if (ihalf < jj) {
+        j = (nleg - jj) + 1;
+        xx = xleg[j-1];
+      } else {
+        j = jj;
+        xx = -xleg[j-1];
+      }
+      var c = b * xx;
+      var ac = a + c;
+
+      // if exp(-qexpo/2) < 9e-14,
+      // then doesn't contribute to integral
+
+      var qexpo = ac * ac;
+      if (qexpo > C3)
+        break;
+
+      var pplus = 2 * jStat.normal.cdf(ac, 0, 1, 1, 0);
+      var pminus= 2 * jStat.normal.cdf(ac, w, 1, 1, 0);
+
+      // if rinsum ^ (cc-1) < 9e-14,
+      // then doesn't contribute to integral
+
+      var rinsum = (pplus * 0.5) - (pminus * 0.5);
+      if (rinsum >= Math.exp(C1 / cc1)) {
+        rinsum = (aleg[j-1] * Math.exp(-(0.5 * qexpo))) * Math.pow(rinsum, cc1);
+        elsum += rinsum;
+      }
+    }
+    elsum *= (((2.0 * b) * cc) / Math.sqrt(2 * Math.PI));
+    einsum += elsum;
+    blb = bub;
+    bub += binc;
+  }
+
+  // if pr_w ^ rr < 9e-14, then return 0
+  pr_w += einsum;
+  if (pr_w <= Math.exp(C1 / rr))
+    return 0;
+
+  pr_w = Math.pow(pr_w, rr);
+  if (pr_w >= 1) // 1 was iMax was eps
+    return 1;
+  return pr_w;
+}
+
+function tukeyQinv(p, c, v) {
+  var p0 = 0.322232421088;
+  var q0 = 0.993484626060e-01;
+  var p1 = -1.0;
+  var q1 = 0.588581570495;
+  var p2 = -0.342242088547;
+  var q2 = 0.531103462366;
+  var p3 = -0.204231210125;
+  var q3 = 0.103537752850;
+  var p4 = -0.453642210148e-04;
+  var q4 = 0.38560700634e-02;
+  var c1 = 0.8832;
+  var c2 = 0.2368;
+  var c3 = 1.214;
+  var c4 = 1.208;
+  var c5 = 1.4142;
+  var vmax = 120.0;
+
+  var ps = 0.5 - 0.5 * p;
+  var yi = Math.sqrt(Math.log(1.0 / (ps * ps)));
+  var t = yi + (((( yi * p4 + p3) * yi + p2) * yi + p1) * yi + p0)
+     / (((( yi * q4 + q3) * yi + q2) * yi + q1) * yi + q0);
+  if (v < vmax) t += (t * t * t + t) / v / 4.0;
+  var q = c1 - c2 * t;
+  if (v < vmax) q += -c3 / v + c4 * t / v;
+  return t * (q * Math.log(c - 1.0) + c5);
+}
+
+jStat.extend(jStat.tukey, {
+  cdf: function cdf(q, nmeans, df) {
+    // Identical implementation as the R ptukey() function as of commit 68947
```

Comment:
Thanks for pointing that out. IANAL so not sure there that line is drawn. I'll look at the source in R and see if there are similar implementations out there that aren't GPL.